diff --git a/src/gui/kernel/qopenglcontext.cpp b/src/gui/kernel/qopenglcontext.cpp
index 8aea593..a1b7586 100644
--- a/src/gui/kernel/qopenglcontext.cpp
+++ b/src/gui/kernel/qopenglcontext.cpp
@@ -979,6 +979,11 @@ bool QOpenGLContext::makeCurrent(QSurface *surface)
         return false;
     }
 
+    QByteArray env = qgetenv("QT_ENABLE_GLYPH_CACHE_WORKAROUND");
+    if (env == "1" || env == "true") {
+        d->workaround_brokenFBOReadBack = true;
+    }
+
     QOpenGLContext *previous = QOpenGLContextPrivate::setCurrentContext(this);
 
     if (d->platformGLContext->makeCurrent(surface->surfaceHandle())) {

